### PR TITLE
Allow for [-oo, oo] ranges for bools

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114362

This fixes a problem in Seamless M4T in fairseq2 repro
instructions at https://docs.google.com/document/d/1PVy4KibfljirQDoijOwyHCV97B67r_iElWqFh7h1Acc/edit

I tried extracting a minimal repro but I couldn't actually manage it!

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/utils/_sympy/value_ranges.py [hunk 1]
```diff
@@ -119,6 +127,10 @@ def is_singleton(self) -> bool:
     def unknown(cls):
         return cls(-sympy.oo, sympy.oo)
 
+    @classmethod
+    def unknown_bool(cls):
+        return cls(sympy.false, sympy.true)
```
Comment (bdhirsh) [inline torch/utils/_sympy/value_ranges.py:132]: It's hard for me to picture what sort of user code + compiler analysis from our side would result in a boolean value_range 🤔 any example / test to go off?

Comment (ezyang) [reply torch/utils/_sympy/value_ranges.py:132]: It is very strange! Maybe I need to study the original problem in more detail

**Pull request closed without merging**